src/jbuild.ml

```diff
@@ -378,6 +378,42 @@ module Preprocess = struct
     | _ -> []
 end
 
+module Blang = struct
+  type 'a t = 'a Blang.t
+
+  open Blang
+  let ops =
+    [ "=", Op.Eq
+    ; ">=", Gte
+    ; "<=", Lt
+    ; ">", Gt
+    ; "<", Lt
+    ; "<>", Neq
+    ]
+
+  let t =
+    let ops =
+      List.map ops ~f:(fun (name, op) ->
+        ( name
+        , (let%map x = String_with_vars.t
+           and y = String_with_vars.t
+           in
+           Compare (op, x, y))))
+    in
+    let t =
+    fix begin fun (t : String_with_vars.t Blang.t Sexp.Of_sexp.t) ->
+      [ "or", repeat t >>| (fun x -> Or x)
+      ; "and", repeat t >>| (fun x -> And x)
+      ] @ ops
+      |> sum ~rest:(String_with_vars.t >>| fun v -> Expr v)
```

Comment:
I'm dubious about this `rest` argument. I feel like it's going to make error messages worse. Could you use `if_list` instead?